### PR TITLE
Remove mysqlxx::Pool::Entry assignment operator

### PR DESCRIPTION
**Pull request by azat**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 86789491be1a945602f6ebf0b3b93bf5272e52ab with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65920/86789491be1a945602f6ebf0b3b93bf5272e52ab/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by antaljanosbenjamin** [comment]
@azat could you please write a few words about why it is beneficial to remove it?

File: src/Common/mysqlxx/mysqlxx/Pool.h [hunk 1]
```diff
@@ -64,17 +64,6 @@ class Pool final
             decrementRefCount();
         }
 
-        Entry & operator= (const Entry & src) /// NOLINT
```
Comment (azat) [inline src/Common/mysqlxx/mysqlxx/Pool.h:67]: This assignment operator may be error-prone, and it is possible for use-after-free, it is not possible in the current usage, but still, imagine that before `incrementRefCount`, `mysqlxx::Pool::tryGet` removes this connection due to `ref_count == 0` - https://github.com/ClickHouse/ClickHouse/blob/76119a4567ce2ac9c0aff715c1a9ba2607e806e0/src/Common/mysqlxx/Pool.cpp#L232-L244

**Review by antaljanosbenjamin — APPROVED** [review]

**Comment by antaljanosbenjamin** [comment]
`00504_mergetree_arrays_rw` - https://github.com/ClickHouse/ClickHouse/pull/66248, merging

**Pull request merged**